### ISSUE TITLE
isBlockInSight() broken?

### ISSUE THREAD

**Issue by PilzAdam** [minetest/minetest#2070]
This doesn't look right:
![isblockinsight_broken](https://cloud.githubusercontent.com/assets/2159710/5631484/6448e4c6-95c7-11e4-93d0-1aa020dfbc3b.png)
(The blocks are loaded, moving a bit shows them again)

**Comment by mamoit** [comment]
It doesn't seem to be isBlockInSight(), since commenting
```
if(cosangle < cos(camera_fov / 2))
    return false;
```
doesn't fix this.

**Comment by kahrl** [comment]
It's caused by occlusion culling.

**Comment by mamoit** [comment]
Then shouldn't [this line](https://github.com/minetest/minetest/blob/master/src/clientmap.cpp#L165) be:
```
if(count > needed_count)
```
Fixes the problem, but may not be the right way to do it.
That secction also needs some commenting :smile:

**Comment by PilzAdam** [comment]
Is this the same bug?
![anotherbug](https://cloud.githubusercontent.com/assets/2159710/6128521/deac73ba-b136-11e4-9a28-09ffbd6a38a7.png)

**Comment by mamoit** [comment]
Try building [my branch](https://github.com/mamoit/minetest/tree/2070-occlusion-fix) and see if that happens, if it doesn't, then it is.

**Comment by PilzAdam** [comment]
@mamoit Still happens with your branch.

**Comment by mamoit** [comment]
I think the occlusion is being calculated based on the position of the character, and not the camera.
If so that is a different issue.
@PilzAdam what happens when you jump on that spot without touching the mouse?

**Comment by PilzAdam** [comment]
@mamoit Yea, that probably causes it. The blocks become visible when jumping.

**Comment by mamoit** [comment]
@PilzAdam want to open another issue with that?
or should I?

**Comment by PilzAdam** [comment]
@mamoit Feel free to open a new issue for this.

**Comment by paramat** [comment]
@PilzAdam still occuring?

**Comment by Fixer-007** [comment]
Was not able to reproduce (tested in different seed). Waiting for others to try.

**Comment by mamoit** [comment]
This was fixed in #2240, but #2307 is still happening.
I didn't close this issue since I don't have permissions to do so.
